tasks/agent.py

```diff
@@ -750,6 +751,10 @@ def version(
     if cache_version:
         create_version_json(ctx, git_sha_length=git_sha_length)
 
+    if release:
+        # Set this environment variable to filter out custom tags that could interfere with the version computation
+        os.environ['RELEASE_CONTEXT'] = 'true'
```

Comment:
Why use an environment variable rather than just adding an argument to `get_version`, `query_version`, and `get_matching_pattern` in `tasks/libs/common/utils.py` ?

Reply:
Because that's already another env variable (`BUCKET_BRANCH`) that triggers the same filter on tags for build of nightly branches. So it was similar and introduced less modifications. I hesitated, but it's probably better to have a real internal variable, yes